src/efftester.ml

```diff
@@ -1089,6 +1185,51 @@ module GeneratorsWithContext (Ctx : Context) = struct
     in
     [ (3, gen) ]
 
+  and option_intro_rules env t eff size =
+    let gen =
+      let open Gen in
+      match t with
+      | Option t' ->
+        Gen.frequencyl [ (2, "Some"); (1, "None") ] >>= fun name ->
+        (match name with
+        | "None" -> return (Some (none t))
+        | "Some" ->
+          list_permute_term_gen_outer env t' eff (size / 3) >>= fun trm ->
```

Comment:
Could you rename `list_permute_term_gen_outer` into `term_gen_sized` globally? This would make the code easier to read; at this point in `option_intro_rules` (for example), it's a breach of abstraction to say "I want to permute lists ...", you just want to generate a term (passing a term and effect, and an explicit size).

Reply:
Here you are making a single sub-call to a term generator, so you don't need to divide the size by 3. I think that `size - 1` would be enough to make sure that the size decreases and is evenly balanced among subterms.